2-dom-manipulation/form.js

```diff
@@ -0,0 +1,80 @@
+const form = document.querySelector('form');
+
+//validate form data
+function validateForm(event) {
+    event.preventDefault();
+    const firstName = document.querySelector('input[id="firstName"]');
+    const lastName = document.querySelector('input[id="lastName"]');
+    const email = document.querySelector('input[id="email"]');
+    const telephone = document.querySelector('input[id="telephone"]');
+    const hobbies = document.querySelectorAll('input[type="checkbox"]');
+    let isHobbySelected = false;
+    let hobbiesSelected = [];
+
+    //regular expresion for name
+    const nameReg = /^[a-zA-Z]+$/;
+
+    //regular expresion for email
+    const emailReg = /^([a-zA-Z0-9_\-\.]+)@([a-zA-Z0-9_\-\.]+)\.([a-zA-Z]{2,5})$/;
+
+    //regular expresion for phone number only numbers
+    const phoneReg = /^[0-9]+$/;
+
+    if(!nameReg.test(firstName.value)) {
+        alert('First name is not valid');
+        return false;
```

Comment:
```suggestion
        return;
```
validate form no está siendo usado cómo una función booleana, puede evitar continuar el flujo dejando `return;`